src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -2023,7 +2024,14 @@ void MergeTreeDataMergerMutator::writeWithProjections(
     std::vector<SquashingTransform> projection_squashes;
     for (size_t i = 0, size = projections_to_build.size(); i < size; ++i)
     {
-        projection_squashes.emplace_back(65536, 65536 * 256);
+        // If the parent part is an in-memory part, squash projection output into one block and
+        // build in-memory projection because we don't support merging into a new in-memory part.
+        // Otherwise we split the materialization into multiple stages similar to the process of
+        // INSERT SELECT query.
+        if (new_data_part->getType() == MergeTreeDataPartType::IN_MEMORY)
+            projection_squashes.emplace_back(std::numeric_limits<size_t>::max(), std::numeric_limits<size_t>::max());
```

Comment:
https://github.com/ClickHouse/ClickHouse/blob/fcfbe4bfea465c056c943b2b2d2a19968e42ce85/src/DataStreams/SquashingTransform.h#L25-L26

Looks like we can use zeroes